lib/routes/routePOST.js

```diff
@@ -4,16 +4,32 @@ import api from '../api/api';
 import routesUtils from './routesUtils';
 import pushMetrics from '../utilities/pushMetrics';
 
+const MAX_POST_LENGTH = 1024 * 1024 * 0.5; // 512 KB
+
 /* eslint-disable no-param-reassign */
 export default function routePOST(request, response, log, utapi) {
     log.debug('routing request', { method: 'routePOST' });
-    request.post = '';
 
+    const post = [];
+    let postLength = 0;
     request.on('data', chunk => {
-        request.post += chunk.toString();
+        post.push(chunk);
+        postLength += chunk.length;
+        // Sanity check on post length
+        if (postLength > MAX_POST_LENGTH) {
+            log.warn('post length is too long',
+                         { postLength });
+            request.pause();
```

Comment:
of these options, the best seems to be to keep the data flowing but stop pushing it (or doing anything with it) once exceed max and then return the error on the on end.  do you agree?